src/gluonts/dataset/parallelized_loader.py

```diff
@@ -346,6 +357,51 @@ def _worker_fn(
     return buf.getvalue()
 
 
+class ShuffleIter(Iterator[DataEntry]):
+    def __init__(
+        self, base_iterator: Iterator[DataEntry], shuffle_buffer_length: int
+    ):
+        self.shuffle_buffer: list = []
+        self.shuffle_buffer_length = shuffle_buffer_length
+        self.base_iterator = base_iterator
+        self.base_iter_finished = False
+
+    def __next__(self) -> DataEntry:
+        # if the buffer is empty, fill the buffer first.
+        # (should only executed in the first round)
+        if not self.shuffle_buffer:
+            self.shuffle_buffer = list(
+                itertools.islice(
+                    self.base_iterator, self.shuffle_buffer_length
+                )
+            )
+        # if buffer still empty, means all elements used,
+        # return a signal of end of iterator
+        if not self.shuffle_buffer:
+            return {}
```

Comment:
I think the right signal to send when the iterator is exhausted is raising a `StopIteration` exception, see here: https://docs.python.org/3/library/stdtypes.html#iterator-types

Reply:
yes, already changed.